Library/Homebrew/linkage_checker.rb

```diff
@@ -191,6 +194,12 @@ def check_dylibs(rebuild_cache:)
           @brewed_dylibs[f] << dylib
         end
       end
+
+      # Binary executables that link @rpath-prefixed dylibs must include at least one LC_RPATH.
+      @executables_missing_rpaths << file if
+        Pathname(file).binary_executable? &&
+        dylibs.any? { |dylib| dylib.start_with? "@rpath/" } &&
+        MachO.open(file).rpaths.empty?
```

Comment:
`MachO` is macOS-only, and I think we only want generic code here. 

We should have a generic `Pathname#rpaths` method you can use instead. This will also avoid the duplicate `MachO.open`, which is pretty expensive.

Reply:
Done.